src/app/cast.js

```diff
@@ -1,64 +1,62 @@
 /**
- * cast module contains the start point to create an Ooyala player and install it on the 
+ * cast module contains the start point to create an Ooyala player and install it on the
  * chromecast receiver app.
- * 
+ *
  * @module cast
  */
-import castManager from './receiverManager'
 import CastPlayer from './player'
 import * as logger from 'loglevel'
 import UIManager from './uiManager'
 import Timer from './idleTimer'
 
 import '../styles/player.css'
 
-var mediaManager = null;
-var castPlayer = new CastPlayer();
+const CASTNAMESPACE = 'urn:x-cast:ooyala';
+const LOG_PREFIX = 'MessageInterceptor:';
+const castPlayer = new CastPlayer();
 
 logger.setLevel(process.env.LOG_LEVEL);
 UIManager.setStatusCast(UIManager.status.LOADING);
-
-// create a new instance of the Media Manager to interact with the sender app request
-mediaManager = new cast.receiver.MediaManager(castPlayer);
-
-// backup of the original onLoad event handler to use it after our custom logic
-mediaManager.origOnLoad = mediaManager.onLoad;
-
-/**
- * onLoad assign a new implementation of the onLoad event that can sets the needed information from the 
- * incoming event
- */
-mediaManager.onLoad = function(e) {
-    logger.info("MediaManager:onLoad", e);
-    castPlayer.setAsset(e.data.media.customData);
-    mediaManager.origOnLoad(e);
-}
-
-
-// backup of the original onPause event handler to use it after our custom logic
-mediaManager.origPause = mediaManager.onPause;
-
-/**
- * onPause overrides the default behaviour to setup the default timeout when an assets it's getting paused 
- */
-mediaManager.onPause = function(e) {
-    logger.info("MediaManager:onPause", e);
+const context = cast.framework.CastReceiverContext.getInstance();
+const playerManager = context.getPlayerManager();
+
+playerManager.setMessageInterceptor(cast.framework.messages.MessageType.LOAD, loadRequestData => {
+    logger.info(LOG_PREFIX, "LOAD", loadRequestData);
+    castPlayer.setAsset(loadRequestData.media.customData);
+    castPlayer.load();
+    playerManager.setMediaElement(castPlayer.OOPlayer);
```

Comment:
I feel that there is no need to set the media element every time the sender tries to load an asset on the receiver. I mean at the end the media element will be always the same. I think we need to find a way to identify the very first time the receiver creates the mediaElement and then pass it to the playerManager.

Let me know your inputs on this.